di-ipv-orchestrator-stub/src/main/java/uk/gov/di/ipv/stub/orc/handlers/IpvHandler.java

```diff
@@ -225,29 +224,37 @@ private URI getIpvBackchannelEndpoint(String environment) throws URISyntaxExcept
                     var accessToken = exchangeCodeForToken(authorizationCode, targetBackend);
 
                     var userInfo = getUserInfo(accessToken, targetBackend);
-
-                    Gson gson = new GsonBuilder().setPrettyPrinting().create();
-                    String userInfoJson = gson.toJson(userInfo);
-                    moustacheDataModel.put("rawUserInfo", userInfoJson);
-
-                    mustacheData = buildMustacheData(userInfo);
-                    moustacheDataModel.put("data", mustacheData);
-                } catch (OrchestratorStubException | ParseException | JsonSyntaxException e) {
-                    List<Map<String, Object>> errorObject =
-                            List.of(Map.of("error_message", e.getMessage()));
-                    moustacheDataModel.put("error", errorObject);
+                    var state = request.queryMap().get("state").value();
+
+                    if (ORCHESTRATOR_STUB_STATE.toString().equals(state)) {
+                        var userInfoJson = gson.toJson(userInfo);
+                        var mustacheData = buildUserInfoMustacheData(userInfo);
+
+                        return ViewHelper.render(
+                                Map.of("rawUserInfo", userInfoJson, "data", mustacheData),
+                                "user-info.mustache");
+                    } else if (AUTH_STUB_STATE.toString().equals(state)) {
+                        return ViewHelper.render(
+                                Map.of("data", userInfo), "mfa-reset-result.mustache");
```

Comment:
Should be be stringifying the `userInfo` blob here too?

Reply:
Can do, only makes it prettier though

Reply:
Everyone loves pretty

Reply:
I need to access the success boolean of it so I might have to make there be multiple params to the template